modules/primo/src/Primo/BriefSearch/Client.php

```diff
@@ -165,4 +124,47 @@ public function validateThumbnail($url) {
     }
   }
 
+  /**
+   * Execute a search.
+   *
+   * @param array $queryParameters
+   *   List of query-parameters on the form [key => [val1, val2...]].
+   *   what options are available.
+   * @param int $offset
+   *   The offset of the search results to return. Use 1 for the first result.
+   * @param int $numResults
+   *   The number of results to return.
+   *
+   * @return \Primo\BriefSearch\Result
+   *   The search result.
+   *
+   * @throws \Primo\Exception\TransferException
+   *   If an error occurs during the execution of the search.
+   */
+  public function search(array $queryParameters, $offset, $numResults) {
+    // Set user-specified "runtime" values.
+    $queryParameters['indx'] = [$offset];
+    $queryParameters['bulkSize'] = [$numResults];
+
+    // Add in "static" defaults.
+    array_walk($this->defaultParameters, function ($param_value, $param_key) use (&$queryParameters) {
+      $queryParameters[$param_key][] = $param_value;
+    });
```

Comment:
Is this actually [`array_merge_recursive`](http://php.net/manual/en/function.array-merge-recursive.php)?